### PR TITLE
feat(api-ts): Report decode error-decodeExpecting

### PR DESCRIPTION
**Pull request by jonball4**
Changes
-
- Add json stringify handling for bigint.
- Report decoding errors encountered in decodeExpecting

Ticket: BG-00000

### CODE REVIEW

**Review by bitgopatmcl — APPROVED** [review]

File: packages/superagent-wrapper/test/request.test.ts [hunk 1]
```diff
@@ -206,7 +206,12 @@ describe('request', () => {
         .then(() => '')
         .catch((err) => (err instanceof DecodeError ? err.message : ''));
 
-      assert.equal(result, 'Could not decode response 200: {"invalid":"response"}');
+      console.log(result);
```
[the remark this answers is not in the document]
Comment (jonball4) [reply packages/superagent-wrapper/test/request.test.ts:209]: whoops - on it.. thanks

[the remark this answers is not in the document]
Comment (jonball4) [reply packages/superagent-wrapper/test/request.test.ts:209]: https://github.com/BitGo/api-ts/pull/269

File: packages/superagent-wrapper/src/request.ts [hunk 2]
```diff
@@ -156,14 +161,14 @@ const patchRequest = <
...
           res.original.body,
         )}`;
         throw new DecodeError(error, res as DecodedResponse<h.HttpRoute>);
       } else if (res.status === 'decodeError') {
-        const error = `Could not decode response ${
-          res.original.status
-        }: ${JSON.stringify(res.original.body)}`;
+        const error = `Could not decode response ${res.original.status}: [${stringify(
+          res.original.body,
+        )}] due to error [${res.error}]`;
```
[the remark this answers is not in the document]
Comment (ericcrosson-bitgo) [reply packages/superagent-wrapper/src/request.ts:171]: While there are more preferable ways to identify a particular error, matching on an `Error`'s message remains prevalent in JavaScript. This change might stop matching a downstream user's code that was written to identify the previous format.

[the remark this answers is not in the document]
Comment (jonball4) [reply packages/superagent-wrapper/src/request.ts:171]: Fair enough, do you object to the change? or are you just pointing out that this should be versioned as a breaking change?

**Pull request merged**